### ISSUE TITLE
package-lock.json is not updated if run with `npx`

### ISSUE THREAD

**Issue by jrencz** [rochejul/npmversion#26]
My use case: I don't have `npmversion` installed in my packages nor expressed in `scripts`.
Instead I use it in a CI job which purpose is to do a version bump and a release.

One of my job steps is `npm i --no-save npmversion && node ./node_modules/.bin/npmversion --increment ${INCREMENT} --preid ${PREID}`

I tried to replace it with `npx npmversion --increment ${INCREMENT} --preid ${PREID}` which theoretically should do the same in one neat expression.
But it doesn't.

The "manual" approach results in 2 files changed: `package.json` and `package-lock.json`.
The npx approach results in only `package.json` being changed unless `package-lock.json` is explicitly specified in `jsonFiles` in `.npmversionrc`

**Comment by rochejul** [comment]
Hi

To achieve that, I run the command "npm version" which does the job to change the  project verson in package.json, package-lock.json and npm-shrinkwrap.json. So even if the command is running with npx, internally, we run npm.

Have you check if there are some issues around calling npm command (like version) through npx ?

I try to have a look on my side

Regards

**Comment by rochejul** [comment]
Weird because running the following command:
> node ./node_modules/npmversion/bin/npmversion --increment patch

It works fine: the package.json and package-lock.json are well updated

Now if I run:
> npx npmversion --increment patch 

Both of the package.json and package-lock.json are updated **BUT** I have an error around the npm error-handler **"npm/lib/utils/error-handler.js:205:18"** for the evaluation of **"if (npm.config.get('json')) {"**

It seems however that the process works and don't prevent to bump and create the commit

My context:

- Windows 10 x64
- Node v8.11.1
- NPM 5.6.0
- NPMVERSION 1.70
- NPX 9.7.1

I try to check that on a linux

Regards

**Comment by jrencz** [comment]
@rochejul I checked on linux Jenkins machine and on a Mac

Node 8.9.1 on mac and 8.11.x on Jenkins
NPM 5.5.1 on mac, 5.6.0 on Jenkins

**Comment by rochejul** [comment]
Ok I try to take some time on the next days to check on my ubuntu (and with nvm) if there is a specific version with an error

Many thanks for your feedback

Regards

**Comment by rochejul** [comment]
Seems to be linked to https://github.com/zkat/npx/issues/143 and to https://github.com/zkat/npx/issues/100

**Comment by rochejul** [comment]
Ok, on my case (for windows), my profile name contains a space (sigh) which produces an error according to https://github.com/zkat/npx/issues/100

Now I have to check why in your case you don't have the expected behavior

Regards

**Comment by rochejul** [comment]
Hummmm very weird because on my ubuntu:
![image](https://user-images.githubusercontent.com/244999/44106649-887209a4-9ff5-11e8-96c1-92d2de61894a.png)

And I have the following configuration:
- node 8.9.1
- npm 5.5.1
- npmversion 1.7.0
- npx 10.2.0

What is your npx version ?

Many thanks

**Comment by jrencz** [comment]
9.6.0 on mac, but I updated and it's the same

```
➜  test-npmversion-npx git:(master) npx --version
9.6.0
➜  test-npmversion-npx git:(master) npm i -g npx
/Users/jrencz/.nvm/versions/node/v8.9.1/bin/npx -> /Users/jrencz/.nvm/versions/node/v8.9.1/lib/node_modules/npx/index.js
+ npx@10.2.0
added 484 packages in 6.423s
➜  test-npmversion-npx git:(master) npx npmversion --increment prepatch --preid npxtest --nogit-commit --nogit-tag
v0.0.3-npxtest.0
➜  pl-fe-ui-angularjs git:(master) ✗ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   package.json

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by jrencz** [comment]
9.6.0 on mac, but I updated and it's the same

```
➜  test-npmversion-npx git:(master) npx --version
9.6.0
➜  test-npmversion-npx git:(master) npm i -g npx
/Users/jrencz/.nvm/versions/node/v8.9.1/bin/npx -> /Users/jrencz/.nvm/versions/node/v8.9.1/lib/node_modules/npx/index.js
+ npx@10.2.0
added 484 packages in 6.423s
➜  test-npmversion-npx git:(master) npx npmversion --increment prepatch --preid npxtest --nogit-commit --nogit-tag
v0.0.3-npxtest.0
➜  test-npmversion-npx git:(master) ✗ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   package.json

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by jrencz** [comment]
I checked one more thing: after analysing npmversion code I found that it's not really npmversion who adjusts the version in `package.json` and `package-lock.json` unless it's directly instructed to via jsonFiles.

I run `npm --no-git-tag-version version 0.0.4` and it updated both files
But then I run `npx npm --no-git-tag-version version 0.0.4` and... it adjusted just `package.json`

So I checked with latest node, npm and npx:

```
➜  text-npmversion-npx git:(master) ✗ nvm i 10.8
######################################################################## 100.0%
Checksums empty
Now using node v10.8.0
➜  text-npmversion-npx git:(master) ✗ npm --version
6.2.0
➜  text-npmversion-npx git:(master) ✗ npm i -g npm
/Users/jrencz/.nvm/versions/node/v10.8.0/bin/npm -> /Users/jrencz/.nvm/versions/node/v10.8.0/lib/node_modules/npm/bin/npm-cli.js
/Users/jrencz/.nvm/versions/node/v10.8.0/bin/npx -> /Users/jrencz/.nvm/versions/node/v10.8.0/lib/node_modules/npm/bin/npx-cli.js
+ npm@6.3.0
added 1 package from 1 contributor and updated 3 packages in 5.869s
➜  text-npmversion-npx git:(master) ✗ npx --version
6.3.0
➜  text-npmversion-npx git:(master) ✗ npm i -g npx
/Users/jrencz/.nvm/versions/node/v10.8.0/bin/npx -> /Users/jrencz/.nvm/versions/node/v10.8.0/lib/node_modules/npx/index.js
+ npx@10.2.0
added 484 packages from 651 contributors in 5.833s
➜  text-npmversion-npx git:(master) ✗ npx npm --no-git-tag-version version 0.0.4
v0.0.4
➜  text-npmversion-npx git:(master) ✗ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   package.json

no changes added to commit (use "git add" and/or "git commit -a")
➜  text-npmversion-npx git:(master) ✗
```

I'm confused

**Comment by jrencz** [comment]
I repeated `npx npm --no-git-tag-version version 0.0.4` on a APFS partition (just default, case insensitive) filesystem and it worked.

I normally work on a case-sensitive Mac OS Extended - there the bug is reproducible. 

I guess it's `npx` bug.
But why does it happen on linux where it's quite normal to have case sensitive filesystems?

**Comment by rochejul** [comment]
@jrencz Yes, npmversion used "npm version" to update package.json, package-lock.json and npm-shrinkwrap.json. So if with npx, it does not work, this is probably an issue on npx side.

I saw some issue around spaces and case sensitives: zkat/npx#100

I keep this issue open until a solution could be found

regards
